jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -216,36 +236,42 @@ def _set_display_unit_of_selected_dataset(self):
         """
 
-        if not self.dataset_selected or not self.aperture_selected:
+        # all cubes are in sb so we can get display unit for plugin from SB display unit
+        # this can be changed to listen specifically to changes in surface brightness
+        # from UC plugin GlobalDisplayUnitChange message, but wiill require some refactoring
+        disp_unit = self.app._get_display_unit('sb')
+
+        # this check needs to be here because 'get_display_unit' will sometimes
+        # return non surface brightness units or even None when the app is starting
+        # up. this can be removed once that is fixed (see PR #3144)
+        if disp_unit is None or not check_if_unit_is_per_solid_angle(disp_unit):
             self.display_unit = ''
             self.flux_scaling_display_unit = ''
             return
 
-        data = self.dataset.selected_dc_item
-        if isinstance(data, list):
-            data = data[0]
-        comp = data.get_component(data.main_components[0])
-        if comp.units:
-            # if data is something-per-solid-angle, its a SB unit and we should
-            # use the selected global display unit for SB
-            if check_if_unit_is_per_solid_angle(comp.units):
-                display_unit_type = 'sb'
-            else:
-                display_unit_type = 'flux'
+        self.display_unit = disp_unit
 
-            disp_unit = self.app._get_display_unit(display_unit_type)
+        # get angle componant of surface brightness
+        # note: could add 'axis=angle' when cleaning this code up to avoid repeating this
 
-            self.display_unit = disp_unit
+        display_solid_angle_unit = check_if_unit_is_per_solid_angle(disp_unit, return_unit=True)
+        if display_solid_angle_unit is not None:
+            self.display_solid_angle_unit = display_solid_angle_unit.to_string()
+        else:
+            # there should always be a solid angle, but i think this is
+            # encountered sometimes when initializing something..
+            self.display_solid_angle_unit = ''
 
-            # now get display unit for flux_scaling_display_unit. this unit will always
-            # be in flux, but it will not be derived from the global flux display unit
-            # note : need to generalize this for non-sr units eventually
-            fs_unit = u.Unit(disp_unit) * u.sr
-            self.flux_scaling_display_unit = fs_unit.to_string()
+        # flux scaling will be applied when the solid angle componant is
+        # multiplied out, so use 'flux' display unit
+        fs_unit = self.app._get_display_unit('flux')
+        self.flux_scaling_display_unit = fs_unit
 
-        else:
-            self.display_unit = ''
-            self.flux_scaling_display_unit = ''
+        # if cube loaded is per-pixel-squared sb (i.e flux cube loaded)
+        # pixel_area should be fixed to 1
+        if self.display_solid_angle_unit == 'pix2':
+            self.disable_pixarea_input = True

```

Comment:
if this is the only thing that affects `disable_pixarea_input`, maybe vue can just handle comparing the unit directly instead of needing an extra traitlet?

Reply:
can i make this a 1 pt tech debt follow up, i tried a checking it directly in vue and it didn't work out of the box